src/tests/dotnet-trace/CollectCommandFunctionalTests.cs

```diff
@@ -0,0 +1,255 @@
+
+            byte[] expected = Encoding.UTF8.GetBytes(ExpectedPayload);
+            Assert.Equal(expected, args.EventStream.ToArray());
+        }
+
+        private static async Task<string[]> RunAsync(CollectArgs config, MockConsole console)
+        {
+            var handler = new CollectCommandHandler();
+            handler.StartTraceSessionAsync = (client, cfg, ct) => Task.FromResult<CollectCommandHandler.ICollectSession>(new TestCollectSession());
+            handler.ResumeRuntimeAsync = (client, ct) => Task.CompletedTask;
+            handler.CursorOperationsSupported = false;
+            handler.CollectSessionEventStream = (name) => config.EventStream;
+            handler.Console = console;
+            handler.IsOutputRedirected = false;
+
+            int exit = await handler.Collect(
+                config.ct,
+                config.cliConfig,
+                config.ProcessId,
+                config.output,
+                config.buffersize,
+                config.providers,
+                config.profile,
+                config.Format,
+                config.duration,
+                config.clrevents,
+                config.clreventlevel,
+                config.name,
+                config.diagnosticPort,
+                config.showchildio,
+                config.resumeRuntime,
+                config.stoppingEventProviderName,
+                config.stoppingEventEventName,
+                config.stoppingEventPayloadFilter,
+                config.rundown,
+                config.dsrouter
+            ).ConfigureAwait(true);
+            if (exit != 0)
+            {
+                throw new InvalidOperationException($"Collect exited with return code {exit}.");
+            }
+            return console.Lines;
+        }
+
+        private static string[] CollectSanitizer(string[] lines)
+        {
+            List<string> result = new();
+            foreach (string line in lines)
+            {
+                if (line.StartsWith("Process        :", StringComparison.Ordinal))
+                {
+                    result.Add("Process        : <PROCESS>");
+                }
+                else
+                {
+                    result.Add(line);
+                }
+            }
+            return result.ToArray();
+        }
+
+        private sealed class TestCollectSession : CollectCommandHandler.ICollectSession
+        {
+            private static readonly byte[] _testBytes = Encoding.UTF8.GetBytes(ExpectedPayload);
+            private readonly MemoryStream _stream = new(_testBytes);
+            public Stream EventStream => _stream;
+            public void Stop() {}
+            public void Dispose() => _stream.Dispose();
+        }
+
+        public static IEnumerable<object[]> BasicCases()
+        {
+            yield return new object[]
+            {
+                new CollectArgs(output: new FileInfo("noProviders")),
```

Comment:
I think it would be fine to hardcode a default output file name and then check for that same name in the console output. I doubt varying it in each test provides much additional bug-finding value.

Reply:
Done